java/src/main/java/org/whispersystems/signalservice/api/SignalServiceMessageSender.java

```diff
@@ -170,18 +170,30 @@ public void sendMessage(SignalServiceAddress recipient, SignalServiceDataMessage
   public void sendMessage(List<SignalServiceAddress> recipients, SignalServiceDataMessage message)
       throws IOException, EncapsulatedExceptions
   {
-    byte[]              content   = createMessageContent(message);
-    long                timestamp = message.getTimestamp();
-    SendMessageResponse response  = sendMessage(recipients, timestamp, content, true);
+    byte[]                  content                 = createMessageContent(message);
+    long                    timestamp               = message.getTimestamp();
+    EncapsulatedExceptions  encapsulatedExceptions  = null;
+
+    try {
+      sendMessage(recipients, timestamp, content, true);
+    } catch (EncapsulatedExceptions e) {
+      encapsulatedExceptions = e;
+    }
 
     try {
-      if (response != null && response.getNeedsSync()) {
+      if (!store.getSubDeviceSessions(localAddress.getNumber()).isEmpty()) {
```

Comment:
Will do as soon as i find the time, most likely tomorrow.

Do you have an idea who uses the PushServiceSocket so that i can verify that it works as intended?